lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystemConfig.java

```diff
@@ -224,6 +242,31 @@ public S3FileSystemConfig setCannedAcl(ObjectCannedAcl objectCannedAcl)
         return this;
     }
 
+    public RetryMode getRetryMode()
+    {
+        return retryMode;
+    }
+
+    @Config("s3.retry-mode")
+    public S3FileSystemConfig setRetryMode(RetryMode retryMode)
+    {
+        this.retryMode = retryMode;
+        return this;
+    }
+
+    @Min(3)
```

Comment:
It has to be greater than 0 as per sdk docs. I just chose 3 because thats the default min in the SDK if `aws.maxAttempts` is not set https://github.com/aws/aws-sdk-java-v2/blob/master/core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/retry/SdkDefaultRetrySetting.java#L91. Im ok with reverting it to 1 if you feel strongly about it.